pulp_container/app/registry_api.py

```diff
@@ -349,6 +321,19 @@ def get_dr_push(self, request, path, create=False):
                 raise RepositoryNotFound(name=path)
         return distribution, repository
 
+    def create_dr(self, path, request):
+        with transaction.atomic():
+            repository = serializers.ContainerPushRepositorySerializer.get_or_create({"name": path})
+            distribution = serializers.ContainerDistributionSerializer.get_or_create(
+                {"base_path": path, "name": path}, {"repository": get_url(repository)}
+            )
+
+            dist_repository = distribution.repository.cast()
```

Comment:
you need to account for the case when distribution.repository will be None

Reply:
Oh, yes.